lib/install/diff-trees.js

```diff
@@ -122,6 +123,8 @@ var diffTrees = module.exports._diffTrees = function (oldTree, newTree) {
   Object.keys(flatOldTree).forEach(function (flatname) {
     if (flatNewTree[flatname]) return
     var pkg = flatOldTree[flatname]
+    if (pkg.isInLink && /^[.][.][/\\]/.test(path.relative(newTree.realpath, pkg.realpath))) return
```

Comment:
Filter out any removes that would happen inside links to modules that aren't inside our project.